x-pack/test/observability_functional/apps/observability/alerts/pagination.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+const ROWS_COUNT_TO_HIDE_PAGE_SELECTOR = 10;
+const DEFAULT_ROWS_PER_PAGE = 50;
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+
+  describe('Observability alerts pagination', function () {
+    this.tags('includeFirefox');
+
+    const retry = getService('retry');
+    const observability = getService('observability');
+
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/observability/alerts');
+      await observability.alerts.common.navigateToTimeWithData();
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    describe(`When less than ${ROWS_COUNT_TO_HIDE_PAGE_SELECTOR} alerts are visible in the screen`, () => {
+      before(async () => {
+        // current archiver has 3 closed alerts
+        await observability.alerts.common.setWorkflowStatusFilter('closed');
+      });
+
+      after(async () => {
+        await observability.alerts.common.setWorkflowStatusFilter('open');
+      });
+
+      it('Does not render page size selector', async () => {
+        await observability.alerts.pagination.missingPageSizeSelectorOrFail();
+      });
+
+      it('Does not render pagination controls', async () => {
+        await observability.alerts.pagination.missingPrevPageButtonOrFail();
+      });
+    });
+
+    describe(`When more than ${ROWS_COUNT_TO_HIDE_PAGE_SELECTOR} alerts are visible in the screen`, () => {
+      before(async () => {
+        // current archiver has 12 open alerts
+        await observability.alerts.common.setWorkflowStatusFilter('open');
+      });
+
+      describe('Page size selector', () => {
+        it('Renders page size selector', async () => {
+          await observability.alerts.pagination.getPageSizeSelectorOrFail();
+        });
+
+        it('Default rows per page selector is 50', async () => {
+          await retry.try(async () => {
+            const defaultAlertsPerPage = await (
+              await observability.alerts.pagination.getPageSizeSelector()
+            ).getVisibleText();
+            expect(defaultAlertsPerPage).to.contain(DEFAULT_ROWS_PER_PAGE);
+          });
+        });
+
+        it('10 rows per page selector works', async () => {
```

Comment:
Nit: Could I suggest re-wording this (and the test below) to something like "it('is possible to show 10 rows per page')"?